modules/auxiliary/admin/dcerpc/icpr_cert.rb

```diff
@@ -195,10 +234,81 @@ def build_csr(cn:, private_key:, dns: nil, msext_upn: nil)
       ))
     end
 
-    request.sign(private_key, OpenSSL::Digest.new('SHA256'))
+    request.sign(private_key, OpenSSL::Digest.new(algorithm))
     request
   end
 
+  # Make a certificate request on behalf of another user.
+  #
+  # @param [OpenSSL::X509::Request] csr The certificate request to make on behalf of the user.
+  # @param [String] on_behalf_of The user to make the request on behalf of.
+  # @param [OpenSSL::X509::Certificate] cert The public key to use for signing the request.
+  # @param [OpenSSL::PKey::RSA] key The private key to use for signing the request.
+  # @param [String] algorithm The digest algorithm to use.
+  # @return [Rex::Proto::Kerberos::Model::Pkinit::ContentInfo] The signed request content.
+  def build_on_behalf_of(csr:, on_behalf_of:, cert:, key:, algorithm: 'SHA256')
```

Comment:
It'd be great to add a quick spec for this method to verify that the `ContentInfo` der is as you expect it - just to catch any shifts in the upstream library

Reply:
Or if it's easier to just integration test the full module, and assert that `store_loot` is called with the right der bytes - that works for me too :+1:

Reply:
Added in 749bfa2aadbd96ca33800d53da34dec640b1a2c2, let me know if that's not what you had in mind.